sweep/sweeper.go

```diff
@@ -441,8 +574,65 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch,
 
 			bestHeight = epoch.Height
 
-			log.Debugf("New blocks: height=%v, sha=%v",
-				epoch.Height, epoch.Hash)
+			log.Debugf("New block: height=%v, sha=%v", epoch.Height,
+				epoch.Hash)
+
+			// In order to correctly meet the confirmation targets
+			// of inputs, we'll bump their fee rates accordingly.
+			//
+			// NOTE: Meeting the confirmation target is a bit
+			// inaccurate at the moment due to the sweeper randomly
+			// rescheduling publish attempts for inputs.
+			for input, confTarget := range s.confTargetInputs {
+				pendInput := s.pendingInputs[input]
+
+				// We'll only proceed to query for a new fee
+				// rate if we don't exceed the threshold of
+				// confirmations left in order to do so.
+				expectedConfHeight := pendInput.initialHeight +
+					confTarget
+				numBlocksLeft := expectedConfHeight - bestHeight
+				if numBlocksLeft < 1 {
+					numBlocksLeft = 1
+				}
+				threshold := int32(
+					float64(confTarget) * confTargetRatio,
+				)
+
+				if numBlocksLeft > threshold {
+					continue
+				}
```

Comment:
I am not sure if we should do this already now. I'd be worried that on average users are going to end up paying more. Is this an actual problem to solve? To me it seems that being able to adjust the conf target or the fee rate directly is enough control.

Suppose the conf target is 6. We published with a fee that should get is confirmed after max 6 blocks. Suppose it would happen after exactly 6 blocks, which is within what is specified. We wouldn't need to do anything. But this logic starts bumping the fee up already after 1 block to the 5 block fee estimate.

Reply:
It's definitely a bit aggressive at the moment, but the functionality itself is still useful. Perhaps a better approach is to only start aggressively bumping after a positive delta (e.g., 2 blocks) from the intended confirmation height?

Reply:
Yes, so I am arguing that it may be too aggressive. If I pay for 6 confs and after 1 block I already become nervous and up the fee, isn't that strange?

We should also not lose sight of the problem to solve. Is it really this strict enforcement of a deadline or do we just want to get some more control over the sweeper process without hard guarantuees?

Reply:
> If I pay for 6 confs and after 1 block I already become nervous and up the fee, isn't that strange?

That's why the threshold exists. The point is to increase the fee rate _only_ if we need to in order to meet the confirmation target. If the fee rate doesn't change, then we won't update it.

The problem to solve is that confirmation targets won't be retried when scheduling/performing a sweep. If fees rise, the sweeper would handle that. With the buckets however, confirmation targets aren't retried. We could keep the behavior the same by querying for the same confirmation target like it previously would, but it's also about enforcing what "confirmation target" actually means. The current behavior doesn't hold the guarantees the term implies.

Reply:
It could be useful to distinguish inputs that we _want_ to be swept within X blocks, and inputs that _has_ to be swept within X blocks (absolute block height).